shap/maskers/_text.py

```diff
@@ -302,8 +284,13 @@ def __add__(self, o):
     def __len__(self):
         return len(self.g)
 
-def merge_score(group1, group2):
+# special_tokens: tokens (such as separator tokens) that should be grouped last
+def merge_score(group1, group2, special_tokens=None):
     score = 0
+    # ensures special tokens are combined last, so 1st subtree is 1st sentence and 2nd subtree is 2nd sentence
+    if special_tokens is not None:
+        if group1[-1].s in special_tokens and group2[0].s in special_tokens:
+            score -= 10000
```

Comment:
My intuition is that the specific value does not matter. Rather, we should be concerned with the relative value compared to the other score adjustments. This is at least 10x larger than any other adjustmennt so it seems likely that it will properly enforce that the two sentences will always be merged last. However, maybe we could consider making it larger if that is a concern?